samples/cpp/ufacedetect.cpp

```diff
@@ -242,41 +234,43 @@ void detectAndDraw( UMat& img, Mat& canvas, CascadeClassifier& cascade,
     putText(canvas, format("OpenCL: %s, fps: %.1f", ocl::useOpenCL() ? "ON" : "OFF", avgfps), Point(50, 30),
             FONT_HERSHEY_SIMPLEX, 0.8, Scalar(0,255,0), 2);
 
-    for( vector<Rect>::const_iterator r = faces.begin(); r != faces.end(); r++, i++ )
```

Comment:
I think this is a more logical and more standardized approach inside OpenCV samples so :+1: from my side